pkg/operator/configobservation/apienablement/observe_runtime_config_test.go

```diff
@@ -231,32 +231,6 @@ func TestGroupVersionsByFeatureGate(t *testing.T) {
 			},
 			expectedGroupVersions: map[configv1.FeatureGateName][]schema.GroupVersion{},
 		},
-		{
-			name:        "intersection of group versions should result in an error",
-			kubeVersion: semver.MustParse("1.31.0"),
-			groupVersionsByFeatureGate: map[configv1.FeatureGateName][]groupVersionByOpenshiftVersion{
-				"DynamicResourceAllocation": {
-					{KubeVersionRange: semver.MustParseRange("< 1.32.0"), GroupVersion: schema.GroupVersion{Group: "resource-a.k8s.io", Version: "v1alpha2"}},
-					{KubeVersionRange: semver.MustParseRange(">= 1.30.0"), GroupVersion: schema.GroupVersion{Group: "resource-a.k8s.io", Version: "v1alpha3"}},
-					{KubeVersionRange: semver.MustParseRange(">= 1.31.0"), GroupVersion: schema.GroupVersion{Group: "resource-b.k8s.io", Version: "v1alpha2"}},
-				},
-			},
-			expectErrors: true,
-		},
-		{
-			name:        "intersection of group versions across feature gates should result in an error",
-			kubeVersion: semver.MustParse("1.31.0"),
-			groupVersionsByFeatureGate: map[configv1.FeatureGateName][]groupVersionByOpenshiftVersion{
-				"DynamicResourceAllocation": {
-					{KubeVersionRange: semver.MustParseRange("< 1.32.0"), GroupVersion: schema.GroupVersion{Group: "resource-a.k8s.io", Version: "v1alpha2"}},
-					{KubeVersionRange: semver.MustParseRange(">= 1.31.0"), GroupVersion: schema.GroupVersion{Group: "resource-b.k8s.io", Version: "v1alpha2"}},
-				},
-				"DRA": {
-					{GroupVersion: schema.GroupVersion{Group: "resource-b.k8s.io", Version: "v1alpha1"}},
-				},
-			},
-			expectErrors: true,
-		},
```

Comment:
A GV associated with two distinct feature gates seems improbable

Reply:
Also seems safe to me if it does happen because we are strictly enabling and never disabling GVs. It's basically a union.